build_save_model.py

```diff
@@ -0,0 +1,117 @@
+'''
+Crated on June 4, 2020
+Author: Joe Tawea
+
+To Do:
+    try-except to check files exist when imorting them
+    try-except to check folder before saving model and df
+'''
+
+import sys
+import pandas as pd
+from sklearn.utils import shuffle
+from datetime import datetime
+
+from tensorflow.keras.layers import Input, Embedding, Dense
+from tensorflow.keras.layers import Flatten, Concatenate
+from tensorflow.keras.models import Model
+from tensorflow.keras.optimizers import SGD
+
+def load_review(review_pkl):
+
+    review_df = pd.read_pickle(review_pkl)
+
+    # (Optional) Scaled down data
+    ct_df = review_df.groupby(['reviewerID']).count().sort_values(by=['overall'])
+    x_users = ct_df[(ct_df.overall>=10)&(ct_df.overall<=28)]['overall']
+    scaled_df = review_df[review_df['reviewerID'].isin(x_users.index)]
+    # Reset review_df to make this code section optional
+    review_df = scaled_df
+
+    # Reset cat.codes for tensorflow
+    review_df['userID'] = scaled_df['reviewerID'].astype('category').cat.codes
+    review_df['productID'] = scaled_df['asin'].astype('category').cat.codes
+
+    return review_df
+
+def load_review(product_pkl):
+
+    product_df = pd.read_pickle(product_pkl)
+
+    return product_df
+
+def build_model(user_id, product_id):
+
+    # Hyper-parameter - Embedding dimension
+    D = 10
+
+    U = len(set(user_id))    # number of users
+    P = len(set(product_id)) # number of products
+
+    u = Input(shape=(1,))
+    p = Input(shape=(1,))
+    u_emb = Embedding(U, D)(u)        # (num_samples, 1, D)
+    p_emb = Embedding(P, D)(p)        # (num_samples, 1, D)
+    u_emb = Flatten()(u_emb)          # (num_samples, D)
+    p_emb = Flatten()(p_emb)          # (num_samples, D)
+    x = Concatenate()([u_emb, p_emb]) # (num_samples, 2D)
+    x = Dense(512, activation='relu')(x)
+    x = Dense(1)(x)
+    model = Model(inputs=(u,p), outputs=x)
+
+    return model
```

Comment:
This function is really well commented. To make it even better, I would say it needs a docstring in its final version.